services/src/main/java/org/keycloak/services/clientpolicy/executor/FullScopeDisabledExecutor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.keycloak.services.clientpolicy.executor;
+
+import org.keycloak.events.Errors;
+import org.keycloak.models.ClientModel;
+import org.keycloak.representations.idm.ClientPolicyExecutorConfigurationRepresentation;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.services.clientpolicy.ClientPolicyContext;
+import org.keycloak.services.clientpolicy.ClientPolicyException;
+import org.keycloak.services.clientpolicy.context.ClientCRUDContext;
+
+/**
+ * Check that switch "fullScopeAllowed" is not enabled for the clients
+ *
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class FullScopeDisabledExecutor implements ClientPolicyExecutorProvider<ClientPolicyExecutorConfigurationRepresentation> {
+
+    @Override
+    public void executeOnEvent(ClientPolicyContext context) throws ClientPolicyException {
+        ClientCRUDContext clientUpdateContext = null;
+        switch (context.getEvent()) {
+            case REGISTERED:
+                clientUpdateContext = (ClientCRUDContext)context;
+                afterRegister(clientUpdateContext.getTargetClient());
+                break;
+            case UPDATE:
+                clientUpdateContext = (ClientCRUDContext)context;
+                beforeUpdate(clientUpdateContext.getTargetClient(), clientUpdateContext.getProposedClientRepresentation());
+                break;
+            default:
+                return;
+        }
+    }
+
+    @Override
+    public String getProviderId() {
+        return FullScopeDisabledExecutorFactory.PROVIDER_ID;
+    }
+
+    private void afterRegister(ClientModel clientModel) {
+        clientModel.setFullScopeAllowed(false);
```

Comment:
Shouldn't this be configurable via auto configure?

Reply:
This is good point. I was thinking about that and did not do it as for the FAPI purpose it is generally useful to have "fullScopeAllowed" as true. On the other hand, someone may prefer to just validate it and be notified if there is an attempt to create client with invalid configuration, so he can fix it by himself instead of rely auto-configuration. So due to this and due the compatiblity with other executors, I've added "auto-configure" . It is enabled by default for the FAPI to ensure security by default.